protocol/proofs/delayed-weth-expansion.md

```diff
@@ -0,0 +1,62 @@
+# Expanded DelayedWETH Capabilities
+
+## Context
+
+The `DelayedWETH` contract acts as a holding contract for the bonded ETH submitted by any
+participant in a `DisputeGame` contract. As of the Granite upgrade, each implementation of the
+`DisputeGame` uses its own `DelayedWETH` contract. Each `DelayedWETH` contract is `Ownable` and is
+subject to a number of safety net actions by the owner address (holding balances from specific
+addresses or removing all ETH held in the `DelayedWETH` contract). `DelayedWETH` is additionally
+subject to the Superchain-wide pause mechanism within the `SuperchainConfig`. Bonds cannot be
+retrieved by game participants if the Superchain pause is active.
+
+## Problem Statement
+
+Existing security mechanisms within the `DelayedWETH` contract are inconvenient and slow down the
+incident response process. Specifically:
+
+- The `DelayedWETH.hold(address target)` function creates an approval from the `target` address to
+  the `owner` address. Since the `target` can simply remove this approval, `hold` MUST be triggered
+  alongside a `transferFrom` in the same transaction. This adds unnecessary complexity to something
+  that could simply be a single transaction.
+- The various security mechanisms inside of `DelayedWETH` can only be triggered by the System Owner
+  which has a slow SLA. The Deputy Guardian can act to trigger the Superchain-wide pause, but this
+  is highly disruptive to the entire Superchain ecosystem.
+
+Additionally, it should be noted that all ETH inside a `DelayedWETH` contract is pooled together
+regardless of which games are using the contract. We currently mitigate this by using one
+`DelayedWETH` contract per game type, but this expands the number of contracts that the System
+Owner and Guardian roles must manage by at least 2 per additional chain. However, the modifications
+required to appropriately address this issue are sufficiently extensive that they are considered
+out of scope for this proposal.
+
+## Propsed Solution
+
+Here we propose some modifications to `DelayedWETH` that help to resolve the two noted limitations:
+
+1. We propose that `DelayedWETH.hold` executes a `transferFrom` automatically to reduce the number
+  of required transactions from 2 to 1 when holding ETH from a single dispute game.
+2. We propose a new `pause` function on the `DelayedWETH` contract itself that pauses *just* the
+  `DelayedWETH` contract and supplements the Superchain-wide pause functionality. This `pause`
+  function would be executable by the Guardian role and could therefore be triggered quickly in
+  case the System Owner is unable to execute another security action in time.
```

Comment:
1. When DelayedWETH is paused, what specifically is paused? I assume all state changing functions except `unpause`, but it might be a smaller set of functions?
2. What "other security action" would the DelayedWETH owner be expected to execute first? The intent of this question is just to make sure the incident response runbook is fleshed out

Reply:
Updated this to answer your questions.